### PR TITLE
Symfony 3 compatibility

### PR DESCRIPTION
**Pull request by jean-gui**
Use Twig_SimpleFilter to fix Symfony 3 compatibility issue. AFAICT that is the only required change.

### CODE REVIEW

**Comment by jean-gui** [comment]
Apparently, Twig 1.12.0 is required for Twig_SimpleFilter to exist. I'm not sure how best to do that. Should I add a new requirement in composer.json, or simply require a more recent version of Symfony?

**Comment by jean-gui** [comment]
ping @thewilkybarkid

**Comment by jean-gui** [comment]
I've fixed travis:
- removed the build against PHP 5.3.3 which lacks SSL support for composer
- forced min version of TwigBundle to 2.3.24 to ensure Twig 1.12 gets used
@thewilkybarkid Could you merge this PR if it looks ok to you?

File: composer.json [hunk 1]
```diff
@@ -16,12 +16,12 @@
     },
     "require": {
         "php": ">=5.3.3",
-        "symfony/framework-bundle": "~2.1",
+        "symfony/framework-bundle": "~2.3",
```
Comment (thewilkybarkid) [inline composer.json]: This doesn't allow for v3, should be `^2.3 || ^3.0`.

File: composer.json [hunk 2]
```diff
@@ -16,12 +16,12 @@
...
         "misd/linkify": "~1.0"
     },
     "require-dev": {
         "phpunit/phpunit": "~4.0",
-        "symfony/twig-bundle": "~2.1"
+        "symfony/twig-bundle": "~2.3,>=2.3.24"
```
Comment (thewilkybarkid) [inline composer.json]: Should be `^2.3.24 || ^3.0`.

Comment (thewilkybarkid) [inline composer.json]: Just seen this that `2.3.4` is required for the Twig version. A `conflicts` entry for `twig/twig` would be better.

**Comment by thewilkybarkid** [comment]
Sorry for not looking at this sooner. If you could address my comments and rebase, I can merge and release a new version.

**Comment by jean-gui** [comment]
Done, although I didn't implement your comment about Twig version 2.3.4 as I don't quite understand it nor the need for this restriction (isn't twig only at version 1.24.x?).

I've also modified https://github.com/jean-gui/linkify-bundle/commit/fecb906d1d6e39d27a1b765b0b33d98d79895eba#diff-e86258c0e7295ac925dfce2368d1f3b5 since tests apparently started to fail recently.

**Comment by thewilkybarkid** [comment]
Merged as 1e9bcf67c29b2aba35ba9d8143bf5ad47d526805.

**Comment by jean-gui** [comment]
Thanks!

**Pull request closed without merging**